lib/ClangImporter/ClangIncludePaths.cpp

```diff
@@ -296,14 +297,119 @@ getLibStdCxxFileMapping(ASTContext &ctx) {
+  llvm::vfs::FileSystem &VFS = ToolChain.getVFS();
+
+  struct {
+    std::string Path;
+    std::string IncludeVersion;
+    std::string LibraryVersion;
+    int MajorVersion;
+  } WindowsSDK;
+  if (llvm::getWindowsSDKDir(VFS, {}, {}, {},
+                             WindowsSDK.Path, WindowsSDK.MajorVersion,
+                             WindowsSDK.IncludeVersion,
+                             WindowsSDK.LibraryVersion)) {
+    llvm::SmallString<261> WinSDKInjection{WindowsSDK.Path};
+    llvm::sys::path::append(WinSDKInjection, "Include");
+    if (WindowsSDK.MajorVersion > 8)
+      llvm::sys::path::append(WinSDKInjection, WindowsSDK.IncludeVersion, "um");
+    llvm::sys::path::append(WinSDKInjection, "module.modulemap");
+
+    AuxiliaryFile = GetWindowsAuxiliaryFile("winsdk.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(WinSDKInjection), AuxiliaryFile);
+  }
+
+  struct {
+    std::string Path;
+    std::string Version;
+  } UCRTSDK;
+  if (llvm::getUniversalCRTSdkDir(VFS, SearchPathOpts.getWinSDKRoot(),
+                                  SearchPathOpts.getWinSDKVersion(), {},
+                                  UCRTSDK.Path, UCRTSDK.Version)) {
+    llvm::SmallString<261> UCRTInjection{UCRTSDK.Path};
+    llvm::sys::path::append(UCRTInjection, "Include", UCRTSDK.Version, "ucrt");
+    llvm::sys::path::append(UCRTInjection, "module.modulemap");
+
+    AuxiliaryFile = GetWindowsAuxiliaryFile("ucrt.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(UCRTInjection), AuxiliaryFile);
+  }
+
+  struct {
+    std::string Path;
+    llvm::ToolsetLayout Layout;
+  } VCTools;
+  if (llvm::findVCToolChainViaCommandLine(VFS, {}, {}, {}, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaEnvironment(VFS, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaSetupConfig(VFS, VCTools.Path, VCTools.Layout) ||
+      llvm::findVCToolChainViaRegistry(VCTools.Path, VCTools.Layout)) {
+    llvm::SmallString<261> VCToolsInjection{VCTools.Path};
+    llvm::sys::path::append(VCToolsInjection, "include");
+
+    llvm::sys::path::append(VCToolsInjection, "module.modulemap");
+    AuxiliaryFile =
+        GetWindowsAuxiliaryFile("vcruntime.modulemap", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(VCToolsInjection), AuxiliaryFile);
+
+    llvm::sys::path::remove_filename(VCToolsInjection);
+    llvm::sys::path::append(VCToolsInjection, "vcruntime.apinotes");
+    AuxiliaryFile =
+        GetWindowsAuxiliaryFile("vcruntime.apinotes", SearchPathOpts);
+    if (!AuxiliaryFile.empty())
+      Mappings.emplace_back(std::string(VCToolsInjection), AuxiliaryFile);
+  }
+
+  return Mappings;
+}
+}
+
 SmallVector<std::pair<std::string, std::string>, 2>
 swift::getClangInvocationFileMapping(ASTContext &ctx) {
   SmallVector<std::pair<std::string, std::string>, 2> result;
 
+  // Linux Mappings
```

Comment:
I would suggest dropping this line because we’re not applying mappings in a platform-based pattern here, so such comment will lead to confusion. Not to say `Glibc` is not Linux-specific (at least in Swift’s world) and we also have C++ stuff that’s required on every platform.

Reply:
The C++ mappings are also platform centric.  libstdc++ vs libc++ vs msvcprt.

Reply:
Ah, I suggested to drop the line because we’re not doing something "grouped by" platforms, which may make some sense to enumerate them in comments.

Another reason is that `Glibc` is not Linux-specific at the moment. It’s (kind of mis-)used for Android, BSD, … and even Cygwin (hypothetically), so it’s incorrect to tag it as "Linux mappings".

Reply:
I think that they are grouped by platforms, but I do see your point about it being applied on the BSDs and Linux.  I can change the comment, but I think that the grouping is important as the injections are platform specific.